src/netex_idf/stops.rs

```diff
@@ -108,6 +237,53 @@ pub fn from_path(path: &std::path::Path, collections: &mut Collections) -> Resul
 mod tests {
     use super::*;
 
+    #[test]
+    #[should_panic(expected = "longitude and latitude not found")]
+    fn test_load_coords_with_one_coord() {
+        let xml = r#"
+<Quay>
+    <Centroid>
+        <Location>
+            <gml:pos srsName="EPSG:2154">666944.0</gml:pos>
+        </Location>
+    </Centroid>
+</Quay>"#;
+        let root: Element = xml.parse().unwrap();
+        load_coords(&root).unwrap();
+    }
+
+    #[test]
+    #[should_panic]
+    fn test_load_unvalid_coords() {
+        let xml = r#"
+<Quay>
+    <Centroid>
+        <Location>
+            <gml:pos srsName="EPSG:2154">666944.0 ABC</gml:pos>
+        </Location>
+    </Centroid>
+</Quay>"#;
+        let root: Element = xml.parse().unwrap();
+        load_coords(&root).unwrap();
+    }
+
+    #[test]
+    #[should_panic]
+    fn test_load_coords() {
+        let xml = r#"
+<Quay>
+    <Centroid>
+        <Location>
+            <gml:pos srsName="EPSG:2154">666944.0 6856019.0</gml:pos>
+        </Location>
+    </Centroid>
+</Quay>"#;
+        let root: Element = xml.parse().unwrap();
+        let coords = load_coords(&root).unwrap();
+
+        assert_eq!((666944.0, 6856019.0), coords);
+    }
+
     #[test]
     fn test_load_stop_areas() {
         let xml = r#"
```

Comment:
Considering the size of this string, shouldn't we move this kind of tests to integration tests instead, with a fixture?

Reply:
yes this will be done when we will more objects parsed